packages/core/src/layout/Page/Page.tsx

```diff
@@ -32,14 +30,17 @@ const useStyles = makeStyles(() => ({
 }));
 
 type Props = {
-  theme?: PageTheme;
+  pageTheme?: PageTheme;
 };
 
-export const Page: FC<Props> = ({ theme = pageTheme.home, children }) => {
+export const Page: FC<Props> = ({
+  pageTheme = customPageTheme.pageTheme.home,
+  children,
+}) => {
   const classes = useStyles();
   return (
-    <PageThemeContext.Provider value={theme}>
+    <ThemeProvider theme={{ pageTheme, ...customPageTheme.baseTheme }}>
```

Comment:
@Rugvip This provider separates main palette. Is this correct approach? (for a while ignore the source which is hardcoded in theme package.)

Reply:
Yep, this is the right idea. The troublesome bit is that we don't have any way to customize the `customPageTheme` being used here. This change is valuable in itself though, and is less complex than figuring out how to customize the page themes. Would be happy the reduce the scope of this PR to just switch out the use `PageThemeContext` with added fields to the `BackstageTheme`.

I'm thinking it makes more sense to add `.pageTheme` or just `.page` to the `BackstageTheme` type rather than creating a new `BackstagePageTheme` type. It's less complex and we can provide some default `.page` theme globally anyway.

Reply:
Btw, theme providers can accept the parent theme to modify it too, e.g.

```ts
theme={theme => ({ ...theme, pageTheme })}
```

Reply:
I'm getting your point here now, Thanks. Let me figure it out and will get back to you soon.

Reply:
@Rugvip Could it be possible?

```tsx
interface BackstageTheme extends Theme {
  palette: BackstagePalette;
  page: {
   entities: Record<string, PageTheme>;
   pagePalette: BackstagePalette;
  };
}
```

Then we can use it at ```Page``` like this.

```tsx
<ThemeProvider theme={theme => ({ theme: pageTheme(will rename it to theme), ...theme.pagePalette })}>
```
I'm not sure about initializing that ```page``` in top level. Happy to get your feedback.

Reply:
> @Rugvip Could it be possible?
> 
> ```tsx
> interface BackstageTheme extends Theme {
>   palette: BackstagePalette;
>   page: {
>    entities: Record<string, PageTheme>;
>    pagePalette: BackstagePalette;
>   };
> }
> ```

I was thinking a bit flatter, where the components inside pages don't need to know how to look up their own theme. Something like this:

```tsx
interface BackstageTheme extends Theme {
  palette: BackstagePalette;
  page: PageTheme;
}
```

That by itself won't allow us to customize the page themes though, but I think it'll take us closer to the end goal. Most importantly it will remove any burden from child elements of a page to have to do any kind of lookup of the theme.

> Then we can use it at `Page` like this.
> 
> ```tsx
> <ThemeProvider theme={theme => ({ theme: pageTheme(will rename it to theme), ...theme.pagePalette })}>
> ```

With the above suggestion this would change to something like this:

```tsx
<ThemeProvider theme={theme => ({ ...theme, page: pageTheme(<something>) })}>
```

Now I'm not sure what `<something>` should be, or where the `pageTheme()` function would come from. We can hardcode both of those just like we have atm, and it would be nice change to get in regardless.

If you want to continue on and explore customizability options I got two options in mind so far:

## Add a PageThemeApi

This would be a Utility API (`packages/core-api/src/apis/definitions`) that is used to select the page theme. Most likely it would expose a single method, something like this:

```tsx
type PageThemeApi = {
  // Entity is the entity we want to theme, pageId is a plain identifier like "tool"
  getPageTheme({ entity, pageId }: { entity: Entity, pageId: string }): PageTheme
}
```

I don't really like this though. Theming is right now configured through the App and not an API. And it feels a bit awkward to be providing parts of the theme through APIs, and parts through the App config. I also added in entity in the example here, but I don't actually think we can do that. Many pages are not tied to entities, and entities is also really a concept that's unknown to the Backstage core, and only added by the catalog plugin. It might actually be nice if we can style based on entities in some cases though, but probably need to skip that.

## Complex MUI Theme Extension

While digging I realized that the MUI theme isn't actually build up of only primitive values, there's actually some more complex functionality tied into the theme itself. For example [getContrastText](https://github.com/mui-org/material-ui/blob/3a60909bbd179d8163cbef572b30dc08de2ddbee/packages/material-ui/src/styles/createPalette.d.ts#L89). Based on that I'm actually thinking that we should just take the above API, remove entity, and just put it on the theme.

This:

```tsx
interface BackstageTheme extends Theme {
  palette: BackstagePalette;
  page: PageTheme;
  getPageTheme: ({ themeId }: { themeId: string }) => PageTheme
}
```

I'd suggest we try this solution out. I prefer the function over a record, because it allows for fallback values to be encoded in a nicer way. It also allows us to add more parameters in the future if we find that a single page theme identifier is too limiting. A downside is that we're not able to enumerate all the page themes, but I think we can live without that.

Reply:
Thanks. I agree with you. Theme extension approach seems good 😼👍

Reply:
@Rugvip Implemented! Please have a look at it.